price_service/server-rust/src/network/rpc.rs

```diff
@@ -0,0 +1,118 @@
+use {
+    crate::{
+        network::p2p::OBSERVATIONS,
+        Vaa,
+    },
+    anyhow::Result,
+    axum::{
+        routing::get,
+        Router,
+    },
+    dashmap::DashMap,
+    std::sync::Arc,
+};
+
+mod rest;
+
+#[derive(Clone, Default)]
+pub struct VaaCache(Arc<DashMap<String, Vec<(i64, Vaa)>>>);
+
+impl VaaCache {
+    /// Add a VAA to the cache. Keeps the cache sorted by timestamp.
+    fn add(&mut self, key: String, timestamp: i64, vaa: Vaa) -> Result<()> {
+        self.remove_expired()?;
+        let mut entry = self.0.entry(key).or_default();
+        let key = entry
+            .binary_search_by(|(t, _)| t.cmp(&timestamp))
+            .unwrap_or_else(|e| e);
```

Comment:
(nitpick: it's a little confusing to call the index into the vec `key` -- i spent a moment trying to figure out why you were redefining the existing map key here. i think `index` would be better)

Also suggest leaving a comment that this is potentially a performance problem -- i think this is doing O(n) work on each insert, and we could do better with a binary search tree. but no need to change it unless there's actually a problem.

Reply:
Yes this wasn't meant to be correct yet, and more demonstrating the structure. I just wrote something that worked, it will improve with the price service implementation.